Dockerfile

```diff
@@ -12,10 +12,13 @@ COPY cmd/    cmd/
 COPY pkg/    pkg/
 
 # Build
+USER root
 RUN CGO_ENABLED=0 GOOS=linux GOFLAGS=-mod=mod go build -a -o manager ./cmd/manager
 
-# Copy the controller-manager into a thin image
-FROM gcr.io/distroless/static:nonroot
+# Use distroless as minimal base image to package the manager binary
+FROM registry.access.redhat.com/ubi8/ubi-minimal:8.6
 COPY third_party/ /third_party/
 COPY --from=builder /go/src/github.com/kserve/kserve/manager /
-ENTRYPOINT ["/manager"]
+USER 65532:65532
```

Comment:
any reason to use such a high number?

it is better to use the same approach than here:
```
RUN microdnf install -y shadow-utils && \
    microdnf clean all && \
    useradd kserve -m -u 1000 -d /home/kserve
```

Reply:
followed the same pattern as the dockerfiles we have for odh-model-controller/modelmesh repos

Reply:
the `useradd` pattern led to some errors. The pattern in this PR (USER root -> build -> set different user at the end) is what other Dockerfiles for our components are following. @Jooho @spolti let me know if this is fine, IMO it's ok.